### DISCUSSION TITLE
Action: Store reference of newly created doc (of other collection) in current row

### DISCUSSION THREAD

**Discussion by webwurm** [AntlerVC/firetable#1077] [Support / Q&A] [answered]
I have a short question and hope I can get some help.

With an action-script I am creating a new document in another collection and would like to store the reference to it in the current row. My code creates the new document, but fails to put the reference in the field. What am I doing wrong?

I tried this:
```
  ...
    db.collection('SearchProfiles').add(newEntry)
    .then((docRef) => {
        // set defaultSearchProfile of User to newly created Ref
        row.defaultSearchProfile = docRef;
      })
    .catch((error) => {
      return {
        success: false,
        message: 'Error adding document: ' + error
      }
    });
  ...
```
I guess my line `row.defaultSearchProfile = docRef;` is wrong - but I can't figure out what it should be...
Thanks!

**Comment by BeeBombshell** [comment, marked as answer]
Hey @webwurm, is `defaultSearchProfile` of the right data type? I suggest changing it to Long Text type and use `JSON.stringify()` to pass the docRef.
```
JSON.stringify(docRef)
```

To test this out, you could use [Rowy Logging](https://docs.rowy.io/cloud-logs) as well.

**Reply by webwurm** [reply]
**Thanks! With logging I figured out 2 things:**
`docRef.id` is better than just `docRef`, because only `docRef `gives me indeed a JSON-payload and otherwise I only get the id

But also I figured out that I can not just set a value of a longText-column like so:
`row.defaultSearchProfile = 'xxx';`

That doesn't work...but I also don't get an error-message. What is the problem here?

**Reply by BeeBombshell** [reply]
I got it... Instead of assigning the value directly, you should use `ref.update` as follows:

```
ref.update({  columnName: await JSON.stringify(docRef)})
```

**Reply by webwurm** [reply]
Thanks, that works perfectly! And to set a reference-field (and not a longText-field) I use the doc.ref(String):
`ref.update({defaultSearchProfile: db.doc('SearchProfiles/' + docRef.id)});`
